packages/next/build/swc/index.js

```diff
@@ -351,7 +376,60 @@ function loadNative() {
             ...options,
             noOpen: options.noOpen ?? true,
           }
-          bindings.startTurboDev(toBuffer(devOptions))
+
+          if (!isCustomTurbopack) {
+            bindings.startTurboDev(toBuffer(devOptions))
+          } else if (!!__INTERNAL_CUSTOM_TURBOPACK_BINARY) {
+            console.warn(
+              `Loading custom turbopack binary from ${__INTERNAL_CUSTOM_TURBOPACK_BINARY}`
+            )
+
+            return new Promise((resolve, reject) => {
+              const spawn = require('next/dist/compiled/cross-spawn')
+              const args = []
+
+              Object.entries(devOptions).forEach(([key, value]) => {
+                let cli_key = `--${key.replace(
+                  /[A-Z]/g,
+                  (m) => '-' + m.toLowerCase()
+                )}`
+                if (key === 'dir') {
+                  args.push(value)
+                } else if (typeof value == 'boolean' && value === true) {
+                  args.push(cli_key)
+                } else if (typeof value != 'boolean' && !!value) {
+                  args.push(cli_key, value)
+                }
+              })
+
+              console.warn(`Running turbopack with args: [${args.join(' ')}]`)
+
+              const child = spawn(__INTERNAL_CUSTOM_TURBOPACK_BINARY, args, {
+                stdio: 'inherit',
+                env: {
+                  ...process.env,
+                },
+              })
+              child.on('close', (code) => {
+                if (code !== 0) {
+                  reject({
+                    command: `${__INTERNAL_CUSTOM_TURBOPACK_BINARY} ${args.join(
+                      ' '
+                    )}`,
+                  })
+                  return
+                }
+                resolve(0)
+              })
+            })
+          } else if (!!__INTERNAL_CUSTOM_TURBOPACK_BINDINGS) {
+            console.warn(
+              `Loading custom turbopack bindings from ${__INTERNAL_CUSTOM_TURBOPACK_BINARY}`
+            )
+            console.warn(`Running turbopack with args: `, devOptions)
+
+            require(__INTERNAL_CUSTOM_TURBOPACK_BINDINGS).startDev(devOptions)
```

Comment:
```suggestion
            require(__INTERNAL_CUSTOM_TURBOPACK_BINDINGS).startTurboDev(toBuffer(devOptions))
```
Is this meant to be a custom `swc/index.js` that wraps loading the bindings again or the napi build itself?

Reply:
Expected usecase is `__INTERNAL_CUSTOM_TURBOPACK_BINDINGS` points to the native napi bindings directly (`.node`), reason this passes `Buffer` for the options instead of plain JS object. It is up to caller's responsibility to prepare bindings to work with suggested interface, either if it's custom wrapped .js or direct call to `.node` binary.

Reply:
I honestly throughly considered possible edge cases for this path, as turbopack itself currently dose not produces napi bindings binaries yet. Potentially there'll be couple of iteration based on turbopack side changes.

Reply:
Yeah can be updated in follow-ups as needed from testing